packages/eslint-config/index.js

```diff
@@ -0,0 +1,26 @@
+import js from "@eslint/js";
+import comments from "@eslint-community/eslint-plugin-eslint-comments/configs";
+import prettier from "eslint-config-prettier";
+import perfectionist from "eslint-plugin-perfectionist/configs/recommended-natural";
+import vitest from "eslint-plugin-vitest";
+import ts from "typescript-eslint";
+
+export default [
+  js.configs.recommended,
+  ...ts.configs.strict,
+  ...ts.configs.stylistic,
+  prettier,
+  perfectionist,
+  comments.configs.recommended,
+  {
+    ...vitest.configs.recommended,
+    rules: {
```

Comment:
Evaluate some rule change not included into the default reccomendend or strict configurations like:
- `"max-lines-per-function": ["error", 200]`
-  `"@typescript-eslint/no-unused-expressions": ["error"]`
-  `"@typescript-eslint/no-unused-vars": ["error", { argsIgnorePattern: "^_" }]`
- `"arrow-body-style": "error"`
- `complexity: "error"` whit default 20 value or a little bit more
- `"guard-for-in": "error"` Less useful if class are forbidden
- `eqeqeq: ["error", "smart"]`
- `"no-bitwise": "error"`
- `"no-eval": "error"`
-  `"no-console": "error"`  A logger must be used instead of standard console.log
- `"no-new-wrappers": "error"`
-  `"no-undef-init": "error"` Stylish but less usefull when immutability will be used
- `"no-param-reassign": "error"`
- `"no-var": "error"`
- `radix: "error"`
- `"prefer-const": "error"`

Other stilystic rules can be added using the plugin [eslint-stylistic](https://github.com/eslint-community/eslint-stylistic) when it will support ESLint 9, formatting rules will be deprecated in `typescript-eslint` like `member-delimiter-style` .

Reply:
> * `"max-lines-per-function": ["error", 200]`

what is the rationale behind `200` ? imho we can restrict to 50 or 100

> * "@typescript-eslint/no-unused-vars": ["error", { argsIgnorePattern: "^_" }]

I would remove `argsIgnorePattern` option, and set `args: "after-used"` as it explained [here](https://eslint.org/docs/latest/rules/no-unused-vars#args). In this way we can write 

```js
app.get((req, res) => {
  res.json("hello!");
}
```

instead the ugly 

```js
app.get((_req, res) => {
  res.json("hello!");
}
```

without getting any warning. If also `res` is not used, the linter will raise an `error` as expected. 
Are you ok with this?

> * "arrow-body-style": "error"

I don't understand what you want to achieve with this rule. Would you disable arrow function with `() => {}`  shape or only when it is not needed?

> * eqeqeq: ["error", "smart"]

Why you prefer `smart` over `always`?

>

Reply:
> what is the rationale behind 200 ? imho we can restrict to 50 or 100
In my experience this rule can be hard to follow when the max line number is small. 50 or 100 lines per function produce a good quality and readable code, 200 avoid to write almost unreadable functions. Any team can force a smaller value at project level.

> I would remove argsIgnorePattern option, and set args: "after-used" 
I'll prefer the more strict version to avoid dangerous patter when the parameters share some prop o methods. Is more evident an error when the developer use the wrong argument.

```javascript
(function(foo, bar) {
   // Both foo and bar have a do method
    return bar.do();
})();
```
> - arrow-body-style": "error"

"as-neded". Avoid the `return` when you have only one instruction
```javascript
// bad
const foo = () => {
    return 0;
};
// ok
let foo = () => 0;

// bad
let bar = () => {
    return {
       bar: {
            foo: 1,
            bar: 2,
        }
    };
};
// ok
let bar = () => ({
       bar: {
            foo: 1,
            bar: 2,
        }
    });
```
> - eqeqeq
Why you prefer smart over always?

Is only a less more strict, this rules protect from bad comparison, smart is enough to reach that goal. [ref](https://eslint.org/docs/latest/rules/eqeqeq#smart)

Reply:
style vs safe

Should be enough the highlighting provided by the IDE, it's a good compromise 👍